controllers/packetmachine_controller.go

```diff
@@ -303,24 +325,41 @@ func (r *PacketMachineReconciler) reconcile(ctx context.Context, machineScope *s
 
 func (r *PacketMachineReconciler) reconcileDelete(ctx context.Context, machineScope *scope.MachineScope, clusterScope *scope.ClusterScope, logger logr.Logger) (ctrl.Result, error) {
 	logger.Info("Deleting machine")
+
 	packetmachine := machineScope.PacketMachine
 	providerID := machineScope.GetInstanceID()
+	defaultTags := defaultTags(machineScope.Namespace(), machineScope.Name(), clusterScope.Name())
+
 	if providerID == "" {
-		logger.Info("no provider ID provided, nothing to delete")
+		logger.Info("no provider ID provided, checking for device by tags")
+
+		device, err := r.PacketClient.GetDeviceByTags(machineScope.PacketCluster.Spec.ProjectID, defaultTags)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		if device != nil {
+			if err := r.deleteDevice(device.ID, force); err != nil {
```

Comment:
Did you want to put the same `http.StatusNotFound` check here, or make `deleteDevice` include that logic?

Reply:
I don't think it should be needed here. If by chance the device is deleted in between the list call made in `GetDeviceByTags` and the call to delete, then we'll just end up doing another reconciliation to remove the finalizer.